src/libtrx/game/objects/general/drawbridge.c

```diff
@@ -111,4 +127,87 @@ static void M_Setup(OBJECT *const obj)
     obj->floor_height_func = M_GetFloorHeight;
 }
 
+void M_DropStack(const ITEM *const item)
+{
+    VECTOR *pos = Vector_Create(sizeof(XYZ_32));
+    Drawbridge_GetSectorPositions(item, pos);
+    for (int32_t i = 0; i < pos->count; i++) {
+        MovableBlock_DropStack(
+            item->pos.y, *(const XYZ_32 *)Vector_Get(pos, i), item->room_num);
+    }
+    Vector_Free(pos);
+    pos = nullptr;
+}
+
+static int32_t floor_div(int32_t v, int32_t d)
+{
+    return (v >= 0) ? v / d : -((-v + d - 1) / d);
+}
+
+BOUNDS_16 M_RotateBounds(const BOUNDS_16 bounds, int16_t rot_y)
+{
+    BOUNDS_16 rot_bounds = { 0 };
+
+    switch (rot_y) {
+    case 0:
+    default:
+        rot_bounds = bounds;
+        break;
+    case DEG_90:
+        rot_bounds.min.x = bounds.min.z;
+        rot_bounds.max.x = bounds.max.z;
+        rot_bounds.min.z = -bounds.max.x;
+        rot_bounds.max.z = -bounds.min.x;
+        break;
+    case -DEG_180:
+        rot_bounds.min.x = -bounds.max.x;
+        rot_bounds.max.x = -bounds.min.x;
+        rot_bounds.min.z = -bounds.max.z;
+        rot_bounds.max.z = -bounds.min.z;
+        break;
+    case -DEG_90:
+        rot_bounds.min.x = -bounds.max.z;
+        rot_bounds.max.x = -bounds.min.z;
+        rot_bounds.min.z = bounds.min.x;
+        rot_bounds.max.z = bounds.max.x;
+        break;
+    }
+    return rot_bounds;
+}
```

Comment:
This is largery a repetition of `Item_RotateBounds`. Would be cool to see this extracted somewhere, maybe `math/` module considering how it's not inherently related to items.